@mrene Thanks for the additions.
When I try to build that on NixOS, I get:
```
❯ nix profile --show-trace add -f $NIXPKGS --print-build-logs zlequalizer
zlequalizer> Running phase: unpackPhase
zlequalizer> unpacking source archive /nix/store/xgs0klmxi84a99jy3dm281mwfqbkfjrr-source
zlequalizer> source root is source
zlequalizer> Running phase: patchPhase
zlequalizer> Running phase: updateAutotoolsGnuConfigScriptsPhase
zlequalizer> Running phase: configurePhase
zlequalizer> cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/share/doc/zlequalizer -DCMAKE_INSTALL_INFODIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/strip -DCMAKE_RANLIB=/nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/ranlib -DCMAKE_AR=/nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/s1kkyzcdxbhavq8nc48822wzxv46fs82-zlequalizer-1.0.0 -DKFR_ARCHS:STRING=sse2\;avx\;avx2 -DZL_JUCE_COPY_PLUGIN:STRING=FALSE
zlequalizer> -- Could NOT find Git (missing: GIT_EXECUTABLE)
zlequalizer> Hint: The project() command has not yet been called.  It sets up system-specific search paths.
zlequalizer> -- Failed to determine VERSION from Git tags. Using default version "0.0.0".
zlequalizer> -- Failed to determine HASH from Git logs. Using default hash "0000000".
zlequalizer> -- Major version: 0
zlequalizer> -- The C compiler identification is Clang 21.1.2
zlequalizer> -- The CXX compiler identification is Clang 21.1.2
zlequalizer> -- Detecting C compiler ABI info
zlequalizer> -- Detecting C compiler ABI info - failed
zlequalizer> -- Check for working C compiler: /nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/clang
zlequalizer> -- Check for working C compiler: /nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/clang - broken
zlequalizer> CMake Error at /nix/store/5bn5f4ivqf4xn19khh4kcg4ngnjs6spg-cmake-4.1.2/share/cmake-4.1/Modules/CMakeTestCCompiler.cmake:67 (message):
zlequalizer>   The C compiler
zlequalizer>
zlequalizer>     "/nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/clang"
zlequalizer>
zlequalizer>   is not able to compile a simple test program.
zlequalizer>
zlequalizer>   It fails with the following output:
zlequalizer>
zlequalizer>     Change Dir: '/build/source/build/CMakeFiles/CMakeScratch/TryCompile-ikv1AH'
zlequalizer>
zlequalizer>     Run Build Command(s): /nix/store/asj4vdlgiqpcqc45lq29r4hv4wlfbl63-ninja-1.13.1/bin/ninja -v cmTC_e71c2
zlequalizer>     [1/2] /nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/clang    -MD -MT CMakeFiles/cmTC_e71c2.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_e71c2.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_e71c2.dir/testCCompiler.c.o -c /build/source/build/CMakeFiles/CMakeScratch/TryCompile-ikv1AH/testCCompiler.c
zlequalizer>     [2/2] : && /nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/clang   CMakeFiles/cmTC_e71c2.dir/testCCompiler.c.o -o cmTC_e71c2   && :
zlequalizer>     FAILED: [code=1] cmTC_e71c2
zlequalizer>     : && /nix/store/al7l1c2nwk9c9cicp3l7jfjxi1vilp93-clang-wrapper-21.1.2/bin/clang   CMakeFiles/cmTC_e71c2.dir/testCCompiler.c.o -o cmTC_e71c2   && :
zlequalizer>     /nix/store/416ykpc2bksb90sd1ia8cybxb3p83mrd-binutils-2.44/bin/ld: cannot find -lX11: No such file or directory
zlequalizer>     clang: error: linker command failed with exit code 1 (use -v to see invocation)
zlequalizer>     ninja: build stopped: subcommand failed.
zlequalizer>
zlequalizer>
zlequalizer>
zlequalizer>
zlequalizer>
zlequalizer>   CMake will not be able to correctly generate this project.
zlequalizer> Call Stack (most recent call first):
zlequalizer>   CMakeLists.txt:39 (project)
zlequalizer>
zlequalizer>
zlequalizer> -- Configuring incomplete, errors occurred!
```
Not sure why, and since it's bedtime for me, I won't be able to investigate for a while.

--- a/pkgs/by-name/zl/zlequalizer/package.nix
+++ b/pkgs/by-name/zl/zlequalizer/package.nix
@@ -1,0 +1,120 @@
+{
+  lib,
+  clangStdenv,
+  fetchFromGitHub,
+
+  # nativeBuildInputs
+  cmake,
+  darwin,
+  ninja,
+  pkg-config,
+  writableTmpDirAsHomeHook,
+
+  # buildInputs
+  alsa-lib,
+  curl,
+  expat,
+  fontconfig,
+  freetype,
+  libGL,
+  libXcursor,
+  libXext,
+  libXinerama,
+  libXrandr,
+  libepoxy,
+  libjack2,
+  libxkbcommon,
+  lv2,
+}:
+
+clangStdenv.mkDerivation (finalAttrs: {
+  pname = "zlequalizer";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ZL-Audio";
+    repo = "ZLEqualizer";
+    tag = "${finalAttrs.version}";
+    hash = "sha256-9TmvjBXTrvR0+qnGDFhCczanxiry3d43QVn/pJLUREY=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+    writableTmpDirAsHomeHook
+  ]
+  ++ lib.optionals clangStdenv.hostPlatform.isDarwin [ darwin.sigtool ];
+
+  buildInputs = [
+    curl
+    expat
+    fontconfig
+    freetype
+    lv2
+  ]
+  ++ lib.optionals clangStdenv.hostPlatform.isLinux [
+    alsa-lib
+    libGL
+    libXcursor
+    libXext
+    libXinerama
+    libXrandr
+    libepoxy
+    libjack2
+    libxkbcommon
+  ];
+
+  # JUCE dlopen's these at runtime, crashes without them
+  NIX_LDFLAGS = lib.optionalString clangStdenv.hostPlatform.isLinux (toString [
+    "-lX11"
+    "-lXext"
+    "-lXcursor"
+    "-lXinerama"
+    "-lXrandr"
+  ]);
+
+  # LTO needs special setup on Linux
+  postPatch = lib.optionalString clangStdenv.hostPlatform.isLinux ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
+  '';
+
+  cmakeFlags = [
+    # see: https://github.com/ZL-Audio/ZLEqualizer#clone-and-build
+    (lib.cmakeFeature "KFR_ARCHS" (
+      if clangStdenv.hostPlatform.isAarch64 then "neon64" else "sse2;avx;avx2"
+    ))
+    (lib.cmakeBool "ZL_JUCE_COPY_PLUGIN" false)
+  ];
+
+  installPhase = ''
+    runHook preInstall
+  ''
+  + lib.optionalString clangStdenv.hostPlatform.isLinux ''
+    mkdir -p $out/lib/{lv2,vst3}
+    mkdir -p $out/bin/
+    cp -r "ZLEqualizer_artefacts/Release/LV2/ZL Equalizer 2.lv2" $out/lib/lv2/
+    cp -r "ZLEqualizer_artefacts/Release/VST3/ZL Equalizer 2.vst3" $out/lib/vst3/
+    install -Dm755 "ZLEqualizer_artefacts/Release/Standalone/ZL Equalizer 2" $out/bin/
+  ''
+  + lib.optionalString clangStdenv.hostPlatform.isDarwin ''
+    mkdir -p $out/Applications
+    cp -r ZLEqualizer_artefacts/Release/{AU,VST3} $out/
+    cp -r ZLEqualizer_artefacts/Release/Standalone/* $out/Applications/
+  ''
+  + ''
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://zl-audio.github.io/plugins/zlequalizer2/";
+    description = "Versatile equalizer plugin for VST3, LV2 and standalone";
+    license = lib.licenses.agpl3Plus;
+    maintainers = with lib.maintainers; [
+      magnetophon
+    ];
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
+  };
+})

